example-deployment/deploy.sh

```diff
@@ -88,6 +88,32 @@ kubectl create configmap reporting-server-config --from-file=reporting_server_co
 echo 'deploying reporting-server...'
 kubectl apply -f k8s/reporting-server.yaml
 
+echo 'building reporting-server migrations...'
+docker build -t rmf-web/reporting-server-migrations -f docker/reporting-server-migrations.dockerfile $rmf_web_ws
+echo 'publishing reporting-server-migrations image...'
+docker save rmf-web/reporting-server-migrations | bash -c 'eval $(.bin/minikube docker-env) && docker load'
+
+
+# Checks if the reporting-server and the reporting-server-db are ready
+while [ "$(kubectl get pods -l=app='reporting-server' -o jsonpath='{.items[*].status.containerStatuses[0].ready}')" != "true true" ]; do
+   sleep 5
+   echo "Waiting for the reporting-server to be ready."
+done
+
+# run migration job
+echo 'running reporting-server-migrations-job...'
+.bin/minikube kubectl -- apply -f k8s/jobs/reporting-server-migrations-job.yaml
+
+# if the migration is finished kill the job
+until kubectl get jobs reporting-server-migrations-job -o jsonpath='{.status.conditions[?(@.type=="Complete")].status}' | grep True ;
+do 
+  echo "wait for migration job to finish"
+  sleep 1; 
+done
+
+echo 'killing reporting-server-migrations-job...'
+.bin/minikube kubectl  -- delete -f k8s/reporting-server-migrations-job.yaml
```

Comment:
Why do we need to manually kill the jobs, can't we let it run to completion?

Reply:
So, this job creates a pod which runs the migrations, when the migrations are applied correctly, the job change to a completed state and the migration pod remains between the pods.  

The job and its pod are only deleted if the task was completed successfully.

To answer your question, sure, we can leave that job and pod as completed and no delete them.

Reply:
> So, this job creates a pod which runs the migrations, when the migrations are applied correctly, the job change to a completed state and the migration pod remains between the pods.

> The job and its pod are only deleted if the task was completed successfully.

Sorry, I got lost in these. The job and its pod is deleted when it is completed successfully, and when the migration is completed successfully, the task is completed so the job and pods should be deleted?

If I understand it correctly, the job and pods are left in kubernetes when the migration/job fails? Was this by kubernetes design or some side effect of the migration scripts? Maybe kubernetes designed it this way to give admins a chance to check the logs etc?

Reply:
Oh sorry,


Yeah, this is the default behavior of Kubernetes: If you run a job in Kubernetes, depending on the job a pod is created or not. 

When that job is completed (successfully), that job and its pod are reachable. 
If that job fails, you can configure Kubernetes to retry to run that job (one new job and one new pod).

Those pods and jobs are left on purpose in case you want to see the logs

--------------------------------------------------------------------------------
Our script only deletes the completed jobs and their pod. But we can remove behavior



Is this ok?

Reply:
I'm sure kubernetes has good reasons to keep the pods after completion, I don't think we should try to change that behavior. I assume kubernetes is clever enough to clean up these pods eventually so they don't take up all the disk space?